pkg/basicstation/cups/server.go

```diff
@@ -0,0 +1,283 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cups
+
+import (
+	"context"
+	"crypto"
+	"crypto/tls"
+	"crypto/x509"
+	"net"
+	"net/url"
+	"strings"
+
+	echo "github.com/labstack/echo/v4"
+	"go.thethings.network/lorawan-stack/pkg/component"
+	"go.thethings.network/lorawan-stack/pkg/errors"
+	"go.thethings.network/lorawan-stack/pkg/rpcmetadata"
+	"go.thethings.network/lorawan-stack/pkg/ttnpb"
+	"go.thethings.network/lorawan-stack/pkg/types"
+	"go.thethings.network/lorawan-stack/pkg/web"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+// ServerConfig is the configuration of the CUPS server.
+type ServerConfig struct {
+	ExplicitEnable  bool `name:"require-explicit-enable" description:"Require gateways to explicitly enable CUPS"`
```

Comment:
CUPS is a necessary feature for `BasicStation`gateways. In fact the very first line of the spec is
> Station periodically queries CUPS for updates.

Why is this optional?

PS: This comment is applicable to all functions related to explicit enabling.

Reply:
Not all gateways are BasicStation gateways. This setting (which is false by default btw) simply requires that gateway owners explicitly enable CUPS before their gateways can use it.

Reply:
Yes the CUPS Service is specific to `BasicStation`. Also for `TTG`, this is default as well.